### ISSUE TITLE
[REVIEW]: LATTE: Lightcurve Analysis Tool for Transiting Exoplanets

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2101]
**Submitting author:** @noraeisner (<a href="http://orcid.org/0000-0002-9138-9028">N. L. Eisner</a>)
**Repository:** <a href="https://github.com/noraeisner/LATTE" target ="_blank">https://github.com/noraeisner/LATTE</a>
**Version:** v1.0.0
**Editor:** @xuanxu
**Reviewer:** @christinahedges, @aureliocarnero
**Archive:** <a href="https://doi.org/10.5281/zenodo.3831586" target="_blank">10.5281/zenodo.3831586</a>

## Status

[![status](https://joss.theoj.org/papers/216060523b5767f7d2d7c05c4e2257bc/status.svg)](https://joss.theoj.org/papers/216060523b5767f7d2d7c05c4e2257bc)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/216060523b5767f7d2d7c05c4e2257bc"><img src="https://joss.theoj.org/papers/216060523b5767f7d2d7c05c4e2257bc/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/216060523b5767f7d2d7c05c4e2257bc/status.svg)](https://joss.theoj.org/papers/216060523b5767f7d2d7c05c4e2257bc)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/noraeisner/LATTE" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@christinahedges & @aureliocarnero, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @xuanxu know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/noraeisner/LATTE">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@noraeisner) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/noraeisner/LATTE">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@noraeisner) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @christinahedges, @aureliocarnero it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02101/joss.02101/10.21105.joss.02101.pdf)

**Comment by xuanxu** [comment]
@christinahedges, @aureliocarnero, @noraeisner: this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#2101` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package. 

We aim for reviews to be completed within about 2-4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ask me (or the submitting author, or each other) if you have any questions :)

**Comment by danielskatz** [comment]
@xuanxu - be aware that being in an active collaboration is a conflict - perhaps this review can be completed before the collaboration between @christinahedges and the authors starts

**Comment by xuanxu** [comment]
@danielskatz Agreed. By her comment I understood that she is not collaborating yet and I assumed there was time enough for the review to finish before the visit, but you are right that's just an assumption I made. @christinahedges can you please clarify this point?

**Comment by christinahedges** [comment]
Hi @xuanxu and @danielskatz, yes I am not currently collaborating with Norah, our collaboration starts in a few months. If you're happy this is not a conflict then I'm happy! 👍

**Comment by danielskatz** [comment]
sounds good to me

**Comment by xuanxu** [comment]
👍

**Comment by aureliocarnero** [comment]
Dear @noraeisner, I'm reviewing the installation instructions, and following your recommendations, I came to a message error telling me that:

ModuleNotFoundError: No module named 'tkinter'

It was easy to solve it, I need to do a: "sudo apt-get install python3-tk", but I think it would be nice to explicitly say that in the documentation, same as you name with the matplotlib version: pip install matplotlib==3.2.0rc1
Put another note on tkinter.

**Comment by aureliocarnero** [comment]
Also, in the installation instructions, when you say that "The first time that the code is run it will also have to download the text data files from MAST (this only has to run once)."
Could you please state the size of the data that will be download? Which is ~325M
Thanks  @noraeisner

**Comment by noraeisner** [comment]
Dear @aureliocarnero, thank you so much for reviewing the code. I will definitely add the need for tkinter and the size of dowloaded data to the installation instructions. Thank you so much.

**Comment by aureliocarnero** [comment]
Dear @noraeisner, there is a few comments to make. 
In order to validate the documentation, you should explain how to perform the automated test. I see there is a directory called tests. Could you put in the code documentation how to run the tests? Ideally, should be done in automatic mode, like using pytests.

**Comment by aureliocarnero** [comment]
@noraeisner another thing is documentation is lacking Community guidelines, please put those (see above in the checklist about what it means). If you don't know what these are, you can see examples in other JOSS code repositories

**Comment by aureliocarnero** [comment]
@noraeisner

Also in the README.md file, there are some exclamation marks that I think are an error
"""""""""""""""""""""""""""""""
Arguments

NOTE: all of these arguments (except new-path, auto and targetlist, new-data) can be changed as option in the GUI. They are arguments in case the same options wish to be run multiple times and the user therfore wishes to identify them in the command line when the program is executed.

!!!!!! --new-data
""""""""""""""""""""""""""""""""""""""""
There, before --new-data

Also, correct the word therefore (it is written therfore)

**Comment by aureliocarnero** [comment]
@noraeisner also there is now a new setting called model that you do not explain what it is.
Could you please document it?
Cheers!

**Comment by aureliocarnero** [comment]
@noraeisner I'm playing around with the code and there is something I would recommend that you improve.
When I run the diagnostic plots, after clicking DONE, then the GUI closes, but the code still is working in the background, without noticing. Therefore, it is a bit confusing because the user might think the process is finished, when it is not. It still keeps running in the background without notice

**Comment by aureliocarnero** [comment]
Hello, @noraeisner, I opened an issue in the github repo because I found a bug so I think it is better it is registered there.
Cheers
Aurelio

**Comment by noraeisner** [comment]
Hi @aureliocarnero, thank you so much for all your comments. I have added a section in the readme file on how to execute the automated tests (using python3 -m unittest) and I have added a section and a file (CONTRIBUTIONS.md) for the Community guidelines as well as addressed the exclamation marks and spell mistakes in the readme file. I'm so sorry about missing these. 

The modelling feature is unfortunately not quite ready to be distributed as I'm finding that it struggles to run on certain computers and is compiler specific so I have removed the 'model' option entirely until this is better tested.

I haven't been able to reproduce the error of the 'done' button freezing yet but I am currently looking into it. I'm also trying to reproduce the  'self.canvas.get_property("window").set_cursor(cursord[cursor])'  error. Can I ask what operating software you are using so that I can try and track these issues down that way? I have tested the code on Mac and Linux/Fedora 27 only. 

Thank you!

**Comment by aureliocarnero** [comment]
Thank you a lot @noraeisner for the quick response. 
So I cannot reproduce again either the canvas.get_property("window") error any longer! It remains a mystery why I had 2 separate errors in the same window of time... 
I'm using Ubuntu 19.10

I will test again to see if I get an error again. 

To try the new version where the settings "model" is no longer available... Do I have to install the code again, I guess?

**Comment by aureliocarnero** [comment]
@noraeisner two quick comments:
1) Could you please put in the paper, a reference to Kepler? Second paragraph in the paper, I'm missing the reference to Kepler there.

2) Maybe it is a silly thing, May I would recommend to put the Brasseur, Phillip, Fleming, Mullally & White, 2019 as Brasseur et al, 2019.?

3) Please correct the word "eecuting" in the README.md, here:
To check, please try eecuting the following command in from your command line: python -m tess_stars2px -t 261136679

**Comment by noraeisner** [comment]
@aureliocarnero, I'm glad to hear that the problem is no longer occurring but I agree that it is very strange that two errors happened at the same time. I will test the code on an Ubuntu computer in order to see whether the errors are related to the operating system. 

Yes, I have removed the modelling routine from the GitHub repository (and I will do the same for the pip install version) so it should now run without it if is is re-installed. I'm so sorry for the inconvenience. 

For the other comments:
1. I just added the Kepler reference. That should  definitely not have been missing.
2. I completely agree with you that it should just be Brasseur et al, 2019 but as far as I can tell the compile automatically decides how to write the references. Please do let me know if there is a way for me to change it manually. 
3. I have corrected this spelling mistake, thank you for spotting this. 

It has also come to my attention that some computers have an SSL license issue when running the code for the first time. I have added a couple of sentences to the readme explaining how to diagnose and solve this if it does occur.

**Comment by aureliocarnero** [comment]
> For the other comments:
> 
>     2. I completely agree with you that it should just be Brasseur et al, 2019 but as far as I can tell the compile automatically decides how to write the references. Please do let me know if there is a way for me to change it manually.

About this, don't worry then, it must be the reference style of JOSS, they might have the limit to 5 authors before going to et al.

**Comment by aureliocarnero** [comment]
@noraeisner could you please let me know when the LATTE installation in the new version is available in pip install. Maybe already is.

**Comment by aureliocarnero** [comment]
@noraeisner please update here the documentation changes we have done through the review:
https://pypi.org/project/tessLATTE/

**Comment by aureliocarnero** [comment]
@noraeisner I run the unittests and I had this output (says Run 7 tests, with FAILED errors=3)
1) Could you please put in the section: "Perform automated tests" that you must clone the git repository first and then you must be in the repository home directory to run those?

2) Here is the output of running the unittests:

/home/carnero/codes/LATTE/tests/test_BLS.py:57: ResourceWarning: unclosed file <_io.FileIO name='./tests/tic55525572_lc.fits' mode='rb' closefd=True>
  alltime, allflux, allflux_err, all_md, alltimebinned, allfluxbinned, allx1, allx2, ally1, ally2, alltime12, allfbkg, start_sec, end_sec, in_sec, tessmag, teff, srad = utils.download_data(indir,sector, tic, binfac = 5, test = './tests/tic55525572_lc.fits')
/home/carnero/codes/LATTE/LATTE/filters.py:256: RuntimeWarning: invalid value encountered in greater
  out = np.isnan(work) + (myabs(array - work) > (nsig*sigma))
Detrend data prior to BLS analysis... done.
/home/carnero/.local/lib/python3.7/site-packages/matplotlib/backends/backend_gtk3.py:187: DeprecationWarning: Gtk.Widget.set_double_buffered is deprecated
  self.set_double_buffered(True)
/home/carnero/.local/lib/python3.7/site-packages/matplotlib/backends/backend_gtk3.py:343: DeprecationWarning: Gtk.Window.set_wmclass is deprecated
  self.window.set_wmclass("matplotlib", "Matplotlib")
/home/carnero/.local/lib/python3.7/site-packages/matplotlib/backends/backend_gtk3.py:372: DeprecationWarning: Gtk.Widget.size_request is deprecated
  size_request = child.size_request()
.EEE/home/carnero/codes/LATTE/tests/test_plotting.py:59: ResourceWarning: unclosed file <_io.FileIO name='./tests/tic55525572_lc.fits' mode='rb' closefd=True>
  alltime, allflux, allflux_err, all_md, alltimebinned, allfluxbinned, allx1, allx2, ally1, ally2, alltime12, allfbkg, start_sec, end_sec, in_sec, tessmag, teff, srad = utils.download_data(indir,sector, tic, binfac = 5, test = './tests/tic55525572_lc.fits')
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:2979: RuntimeWarning: invalid value encountered in greater
  lkeep = np.isfinite(X1.sum(axis=1)) * (Q==0) * (X1.sum(axis=1)>0)
/home/carnero/codes/LATTE/tests/test_plotting.py:60: ResourceWarning: unclosed file <_io.FileIO name='./tests/tic55525572_tp.fits' mode='rb' closefd=True>
  X1_list, X4_list, oot_list, intr_list, bkg_list, apmask_list, arrshape_list, t_list, T0_list, tpf_filt_list = utils.download_tpf_mast(indir, transit_sec, transit_list, tic, test = './tests/tic55525572_tp.fits')
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:4170: RuntimeWarning: invalid value encountered in less
  mask_dd = (np.array(time_dd) < peak+0.75) & (np.array(time_dd) > peak-0.75)
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:4170: RuntimeWarning: invalid value encountered in greater
  mask_dd = (np.array(time_dd) < peak+0.75) & (np.array(time_dd) > peak-0.75)
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:4184: RuntimeWarning: invalid value encountered in less
  mask_dd = (np.array(time_dd) < peak+0.75) & (np.array(time_dd) > peak-0.75)
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:4184: RuntimeWarning: invalid value encountered in greater
  mask_dd = (np.array(time_dd) < peak+0.75) & (np.array(time_dd) > peak-0.75)
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:4185: RuntimeWarning: invalid value encountered in less
  mask_dd_binned = (np.array(time_dd_binned) < peak+0.75) & (np.array(time_dd_binned) > peak-0.75)
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:4185: RuntimeWarning: invalid value encountered in greater
  mask_dd_binned = (np.array(time_dd_binned) < peak+0.75) & (np.array(time_dd_binned) > peak-0.75)
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:3395: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  plt.subplot(2,gs,g+1) # define the plotting area
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:3418: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  plt.subplot(2,gs,gs+1+g) # define the plotting area
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:3630: RuntimeWarning: invalid value encountered in less
  mask_dd = (np.array(alltime) < peak+2) & (np.array(alltime) > peak-2)
/home/carnero/codes/LATTE/LATTE/LATTEutils.py:3630: RuntimeWarning: invalid value encountered in greater
  mask_dd = (np.array(alltime) < peak+2) & (np.array(alltime) > peak-2)
1   out of    11 
2   out of    11 
3   out of    11 
4   out of    11 
5   out of    11 
6   out of    11 
7   out of    11 
8   out of    11 
9   out of    11 
10   out of    11 
11   out of    11 

 Calculating the Aperture Mask... done.

Waiting on plot...
/usr/lib/python3.7/unittest/case.py:628: ResourceWarning: unclosed file <_io.FileIO name='./tests/tic55525572_tp.fits' mode='rb' closefd=True>
  testMethod()
.Running Pyaneti modelling - this could take a while so be patient...
Traceback (most recent call last):
  File "/home/carnero/codes/LATTE/LATTE/pyaneti_LATTE.py", line 43, in <module>
    import pyaneti as pti #FORTRAN module
ImportError: bad magic number in 'pyaneti': b'\x03\xf3\r\n'
..
======================================================================
ERROR: test_DVreport (tests.test_DVreport.TestDVreport)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/carnero/codes/LATTE/tests/test_DVreport.py", line 61, in test_DVreport
    ldv.LATTE_DV(tic, indir, syspath, transit_list, sectors_all, target_ra, target_dec, tessmag, teff, srad, [0], [0], tpf_corrupt, FFI = False,  bls = False, model = False, mpi = None, test = './tests/')
TypeError: LATTE_DV() missing 1 required positional argument: 'astroquery_corrupt'

======================================================================
ERROR: tests.test_data_download (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_data_download
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/carnero/codes/LATTE/tests/test_data_download.py", line 17, in <module>
    from nose.tools import assert_true
ModuleNotFoundError: No module named 'nose'


======================================================================
ERROR: tests.test_data_products (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_data_products
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/carnero/codes/LATTE/tests/test_data_products.py", line 17, in <module>
    import mock
ModuleNotFoundError: No module named 'mock'


----------------------------------------------------------------------
Ran 7 tests in 36.193s

FAILED (errors=3)

**Comment by aureliocarnero** [comment]
If there are libraries missing, please correct the pip install execution to correctly import the needed libraries.

**Comment by noraeisner** [comment]
Hi @aureliocarnero, I'm so sorry that the tests didn't run - there were some files missing from the GitHub page. I have updated it so that it should now run along with the readme file. I should note that the runtime warnings still appear when the tests are run, however these don't stop the code,  or tests, form completing. I have looked into them and they are a common problem with the 'requests' and 'reportlab', modules as these don't close their files properly. I haven't been able to find a solution to avoid these but please do let me know if they are a big problem and I will try and find a work-around. 

I have also updated the pip install version - now v0.1.13. Thank you so much.

**Comment by xuanxu** [comment]
@christinahedges did you have the opportunity to start your review?

**Comment by aureliocarnero** [comment]
Hello @xuanxu , could you please recompile the paper draft to show the new changes in the manuscript?
Cheers
Aurelio

**Comment by aureliocarnero** [comment]
Dear @noraeisner . Could you please revise the pip3 install routine? After you updated to the new version, I cannot install LATTE correctly. It runs ok, all the dependencies installations, but when I try to run the program, it tells me that "No module named LATTE"
If I install it from the github repo with "pip install .", it works ok.
Cheers
Aurelio

**Comment by xuanxu** [comment]
@aureliocarnero I'll recompile the paper now. All the participants in the review can regenerate it anytime, with the same command I'm going to issue next:

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by aureliocarnero** [comment]
@whedon generate pdf

**Comment by aureliocarnero** [comment]
@xuanxu thank you, I just saw that it is explained above in the instructions. My fault!
Nonetheless, I do not see the new link to the pdf... Maybe it is because it is taking some time to compile?

**Comment by xuanxu** [comment]
@aureliocarnero Yes, it looks like the process is taking too long, I'm going to take a look.

**Comment by xuanxu** [comment]
In the meantime you can take a look here: http://res.cloudinary.com/hju22ue2k/image/upload/v1583058972/gqtl5cd1ykdsx8v1hdfl.pdf (just created using http://whedon.theoj.org/ )

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by xuanxu** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a  dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02101/joss.02101/10.21105.joss.02101.pdf)

**Comment by christinahedges** [comment]
> @christinahedges did you have the opportunity to start your review?

@xuanxu I'm sorry I haven't started my review I've been traveling and haven't been able to get to it. As @aureliocarnero has already reviewed this work, would you prefer me to review another submission in the near future?

**Comment by xuanxu** [comment]
> I'm sorry I haven't started my review I've been traveling and haven't been able to get to it.

No problem!

> As aureliocarnero has already reviewed this work, would you prefer me to review another submission in the near future?

@christinahedges we want all of the papers to be reviewed by (at least) two different reviewers, so please go ahead with this one.

**Comment by christinahedges** [comment]
Hi @xuanxu no problem, I can do that. I need some time to get spun up, do you have a deadline when you need my review by?

**Comment by xuanxu** [comment]
@christinahedges we aim for two-four weeks for have the review but it's an intention, not a hard deadline. And you can comment and provide feedback to the authors as you go.

**Comment by aureliocarnero** [comment]
Just in case you missed this message:

Dear @noraeisner . Could you please revise the pip3 install routine? After you updated to the new version, I cannot install LATTE correctly. It runs ok, all the dependencies installations, but when I try to run the program, it tells me that "No module named LATTE"
If I install it from the github repo with "pip install .", it works ok.

Also, I still have 1 error when I run the unittest. Is this normal?
=================
======================================================================
FAIL: test_nn (LATTE.tests.test_data_download.TestNearestNeighbours)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/carnero/codes/LATTE/LATTE/tests/test_data_download.py", line 50, in test_nn
    self.assertEqual(output[1], [0.0, 6.795727328666717, 10.622509290130298, 15.63959237521102, 16.881930554329756, 17.79841005439404], "nearest neighbour distances are incorrect")
AssertionError: Lists differ: [0.0,[32 chars]130296, 15.63959237521102, 16.88193055432957, [14 chars]4076] != [0.0,[32 chars]130298, 15.63959237521102, 16.881930554329756,[14 chars]9404]

First differing element 2:
10.622509290130296
10.622509290130298

  [0.0,
   6.795727328666717,
-  10.622509290130296,
?                   ^

+  10.622509290130298,
?                   ^

   15.63959237521102,
-  16.88193055432957,
?                  ^

+  16.881930554329756,
?                 + ^

-  17.798410054394076]
?                  ^^

+  17.79841005439404]
?                  ^
 : nearest neighbour distances are incorrect

----------------------------------------------------------------------
Ran 10 tests in 138.972s

FAILED (failures=1)
===================

 
Cheers
Aurelio

**Comment by noraeisner** [comment]
Hi @aureliocarnero , I am so sorry for missing your previous message. I have re-uploaded the pip install version (v0.1.15). I have verified that it works on two computers (Mac and Ubuntu), however, as far as I can tell the previous version also worked for me so please do let me know if the problem persists. I have also changed the unittests so that the different roundings do not raise an error. Thank you so much for looking at the code!

**Comment by aureliocarnero** [comment]
Hello @noraeisner . Now the installation works!
I don't know why, but if I installed it with pip3, it gave me an error, but if I did it with pip, then it worked fine.
Actually, when I installed with pip3, and executed with python3.6, then it run, so it might be a problem with my python 3 version setup.
Basically, if I'm in python 3 enviroment already, running pip install tessLATTE worked perfectly, without need of pip3.

**Comment by aureliocarnero** [comment]
@noraeisner in terms of the unittest, there is something funny. If I set the --new-path somewhere outside the LATTE directory, it gives me an error, but if I put the path somewhere within the LATTE directory, it runs ok. So I would recommend you say something like this. Could you try to replicate this?

For example: when it asks me about the new-path and I put: ./LATTE_output (what is recommended), then it worked, but if I put /home/carnero/codes/tempLATTE, then it failed.

Nonetheless, now the unittest gives no errors at all! That is good.
I will test the mpi example case, and if I can run it, then I will validate all the code.
We are close!
Cheers
Aurelio

**Comment by christinahedges** [comment]
Hi @xuanxu 

> Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

This link has expired for me, would you be able to tell me how I can get a new one? I'm just going through the checklist now!

Thanks!
C

**Comment by danielskatz** [comment]
@arfon needs to re-enable the invitation

**Comment by arfon** [comment]
You're invited again @christinahedges.

**Comment by christinahedges** [comment]
Thanks @arfon.

I've taken a look at LATTE and added my review comments to an issue [here](https://github.com/noraeisner/LATTE/issues/3). @noraeisner and @xuanxu, I have left comments there why some of the boxes are unticked.

I find the biggest comment I have to be the state of the field in the paper needs to be improved. I think that the SPOC validation reports, MIT QLP validation reports, eleanor, and lightkurve are all relevant context that should be mentioned. I also find in general that the documentation could be improved a little to show the workflow of an average user, and discuss what each validation plot is designed to show.

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by aureliocarnero** [comment]
Hello @noraeisner, I have no capabilities at the moment to run the mpi example.... Nonetheless, we can pause now the review. Don't worry about this now. Take care and speak soon!

**Comment by arfon** [comment]
> Hello @noraeisner, I have no capabilities at the moment to run the mpi example.... Nonetheless, we can pause now the review. Don't worry about this now. Take care and speak soon!

Or you're all welcome to continue working on this... The message above was designed to set expectations for reviewers, editors, and authors accordingly. If you all are able to keep working on this, please feel free to do so.

**Comment by noraeisner** [comment]
Hi all, thank you so much for your review comments and I'm so sorry for my delayed reply. It took me a couple of days to relocate (I'm in isolation in Belgium now) but I will be working on the comments today and I'll respond to the individual comments as I address them. Of course I also completely understand if the reviewers prefer to pause the process for now. I hope everyone is safe and well!

**Comment by aureliocarnero** [comment]
Im glad you are good! We in Spain are isolated as well now... So I'm happy to continue with the validation. Fortunately I'm healthy and I haven't been affected at all so far.
Apart from Christina's comment that I agree with, I just need to look at the parallel example now. I'm happy to continue with the validation until the end....
Let me see if I can finish, my computer at home is not the best though...

**Comment by aureliocarnero** [comment]
Hello @noraeisner so I tested the script example_parallelized_code.py and even though I have MPI installed, I do not have more than 1 core, therefore it enters into the serial part of the code. Here, I have this error. Could you please help me?

Traceback (most recent call last):
  File "example_parallelized_code.py", line 250, in <module>
    transit_list = ast.literal_eval(((df.loc[df['TIC_ID'] == f]['db_peak']).values)[0])
  File "/home/carnero/anaconda3/envs/mypython3/lib/python3.8/site-packages/pandas/core/frame.py", line 2800, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/home/carnero/anaconda3/envs/mypython3/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 2648, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1618, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1626, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'db_peak'

**Comment by noraeisner** [comment]
Hello @aureliocarnero, I'm glad to hear that you are safe and well in Spain! Belgium has now also entered a lock-down which is definitely a strange experience. 

I'm really sorry, the mpi code that was on github before required the user to enter their own file structure within the code itself. I have changed it to make it much more general and it now works with the example input file - I have added a short section to the readme file of how to execute it. 

I also made a small change to the GUI. If more than one sector is now identified, you can chose to look at the sectors one-by-one or all of them together.  I made this change as before it was quite difficult to see what was going on if too many sectors were selected due to the figure dimensions.  Nothing in the code itself has changed except the way that the interface looks.

I have also expanded on the instructions in the readme and added some more background information to the paper (as outlined here https://github.com/noraeisner/LATTE/issues/3).

**Comment by aureliocarnero** [comment]
@noraeisner great improvement! Im my case, we are very near the end of the validation. I'm not English speaker, so in my opinion the paper is fine, but please consider the other referee questions as well. 
Just a final comment, I found a bug in the example_parallelized_code.py code. 
Where it reads:
                utils.data_files(indir)
                utils.tp_files(indir)
                utils.TOI_TCE_files(indir)
                utils.momentum_dumps_info(indir)

I think it should read:

          LATTEutils.data_files(indir)
          LATTEutils.tp_files(indir)
          LATTEutils.TOI_TCE_files(indir)
          LATTEutils.momentum_dumps_info(indir)

Correct? Could you check that for me? And let me know when you commit the change for a final test?

**Comment by arfon** [comment]
:wave: all, just a friendly check-in to see how things are going with this review?

**Comment by noraeisner** [comment]
@aureliocarnero, thank you for seeing that bug, I have changed those lines in example_parallelized_code.py. The final version has been uploaded to Github and the pip version updated (now version 0.1.18) for whenever you have the time. I have addressed Christina comments and am waiting to hear back. Thank you so much.

**Comment by xuanxu** [comment]
Hi @christinahedges @aureliocarnero, hope you are safe and well. Did you have the time to check the latest changes Nora made?

**Comment by aureliocarnero** [comment]
Thank you for the reminder. I will check ASAP.

**Comment by aureliocarnero** [comment]
So @noraeisner I found another error when running both the parallelized example and the unittests. There is an error with download and database issues...

======================================================================
ERROR: test_downloadLC (LATTE.tests.test_data_download.TestDownloadLC)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/carnero/codes/LATTE/LATTE/tests/test_data_download.py", line 71, in test_downloadLC
    alltime, allflux, allflux_err, all_md, alltimebinned, allfluxbinned, allx1, allx2, ally1, ally2, alltimel2, allfbkg, start_sec, end_sec, in_sec, tessmag, teff, srad = LATTEutils.download_data(outdir, [5], '55525572', binfac = 5)
  File "/home/carnero/codes/LATTE/LATTE/LATTEutils.py", line 2639, in download_data
    lchdu  = pf.open(response.url) # this needs to be a URL - not a file
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 150, in fitsopen
    return HDUList.fromfile(name, mode, memmap, save_backup, cache,
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 388, in fromfile
    return cls._readfrom(fileobj=fileobj, mode=mode, memmap=memmap,
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 1039, in _readfrom
    fileobj = _File(fileobj, mode=mode, memmap=memmap, cache=cache)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/utils/decorators.py", line 503, in wrapper
    return function(*args, **kwargs)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/file.py", line 147, in __init__
    self.name = download_file(fileobj, cache=cache)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/utils/data.py", line 1012, in download_file
    with shelve.open(urlmapfn) as url2hash:
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/shelve.py", line 243, in open
    return DbfilenameShelf(filename, flag, protocol, writeback)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/shelve.py", line 227, in __init__
    Shelf.__init__(self, dbm.open(filename, flag), protocol, writeback)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/dbm/__init__.py", line 91, in open
    raise error[0]("db type is {0}, but the module is not "
dbm.error: db type is dbm.gnu, but the module is not available

Could you please check this?
The same message appears when running the mpi example....
Cheers
Aurelio

**Comment by noraeisner** [comment]
@aureliocarnero, I'm so sorry! One of the files went missing in my last git push (my computer had run out of space and started uploading random files to the cloud and off my computer). I've updated the git (unittests only work with the github version as the files that are needed for the tests are large, and it was recommended to keep the pip version as small as possible). I'm so sorry and thank you so much for you patience with this. That should solve the unittest error. If the mpi version still does not work I will remove this functionality for now, if you agree, as it works on both of my computer and I don't know enough about mpi to diagnose the problem on other operating systems (and it's not key to the code by any means). 
Thank you
Nora

**Comment by aureliocarnero** [comment]
@noraeisner no problem! That is why we review it, so it ends up being a clean and neat product = ).
I commit errors all the time.

So, I did a fresh installation, using pip3 install tessLATTE, and I get the error I was getting when I was doing the unittest. I can introduce a TIC ID and sectors, but then, this error message appears:

raceback (most recent call last):
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/LATTE/__main__.py", line 424, in <module>
    utils.interact_LATTE(tic, indir, syspath, sectors_all, sectors, ra, dec, args)  # the argument of whether to shos the images or not 
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/LATTE/LATTEutils.py", line 129, in interact_LATTE
    alltime, allflux, allflux_err, all_md, alltimebinned, allfluxbinned, allx1, allx2, ally1, ally2, alltime12, allfbkg, start_sec, end_sec, in_sec, tessmag, teff, srad = download_data(indir, sectors, tic)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/LATTE/LATTEutils.py", line 2639, in download_data
    lchdu  = pf.open(response.url) # this needs to be a URL - not a file
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 150, in fitsopen
    return HDUList.fromfile(name, mode, memmap, save_backup, cache,
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 388, in fromfile
    return cls._readfrom(fileobj=fileobj, mode=mode, memmap=memmap,
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 1039, in _readfrom
    fileobj = _File(fileobj, mode=mode, memmap=memmap, cache=cache)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/utils/decorators.py", line 503, in wrapper
    return function(*args, **kwargs)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/io/fits/file.py", line 147, in __init__
    self.name = download_file(fileobj, cache=cache)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/site-packages/astropy/utils/data.py", line 1012, in download_file
    with shelve.open(urlmapfn) as url2hash:
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/shelve.py", line 243, in open
    return DbfilenameShelf(filename, flag, protocol, writeback)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/shelve.py", line 227, in __init__
    Shelf.__init__(self, dbm.open(filename, flag), protocol, writeback)
  File "/home/carnero/anaconda3/envs/testLATTE/lib/python3.8/dbm/__init__.py", line 91, in open
    raise error[0]("db type is {0}, but the module is not "
dbm.error: db type is dbm.gnu, but the module is not available

**Comment by noraeisner** [comment]
Thank you! I just looked into that error message and it appears to be a problem with the astropy stored caches (https://github.com/astropy/astropy/issues/9607) so I have added the following two lines to the start of the code in LATTteutils.py:

from astropy.utils.data import clear_download_cache
clear_download_cache()

I can't reproduce the same error on my end so I, unfortunately, can't check whether that works but if you could try it that would be amazing (I have updated the git version or you could copy and paste those two lines into the to the top of LATTEutils.py. Thank you

**Comment by aureliocarnero** [comment]
@noraeisner everything working now.
Congratulations! Great work.
I have validated everything and in my opinion, is ready for publication.

**Comment by noraeisner** [comment]
I'm so glad that that worked and thank you so much for all your feedback and help with fixing these bugs :).

Thanks, 
Nora

**Comment by christinahedges** [comment]
@noraeisner thank you for addressing my comments, the documentation looks great, and the diagnostic plots are very helpful. This is ready for publication.

**Comment by xuanxu** [comment]
Thank you @aureliocarnero and @christinahedges!

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02101/joss.02101/10.21105.joss.02101.pdf)

**Comment by xuanxu** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1117/1.JATIS.1.1.014003 is OK
- 10.3847/1538-3881/aa9be0 is OK
- 10.1111/j.1365-2966.2011.19932.x is OK
- 10.1111/j.1365-2966.2004.07657.x is OK
- 10.3847/1538-3881/aabc4f is OK
- 10.1117/12.2233418 is OK
- 10.1086/667698 is OK
- 10.1093/mnras/sty2472 is OK
- 10.1051/0004-6361/201833051 is OK
- 10.1088/1538-3873/ab291c is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by xuanxu** [comment]
@noraeisner I've issued a pull request in your LATTE repo fixing some typos in the paper text. Can you take a look at it?

**Comment by noraeisner** [comment]
@xuanxu Thank you so much for fixing the typos. I have merged the pull request.

**Comment by xuanxu** [comment]
Thanks @noraeisner! Everything looks ready now. 
At this point could you make a new release of this software that includes the changes that have resulted from this review?. Then, please make an archive of the software in Zenodo (or a similar service) For the Zenodo archive make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

Once you do that please report here the new version and the DOI of the archive.

I can then move forward with accepting the submission.

**Comment by noraeisner** [comment]
@xuanxu I have upload the new release, v1.0.0, to Zenodo under the DOI 10.5281/zenodo.3831586. Thank you!

**Comment by xuanxu** [comment]
Great, thanks @noraeisner!

**Comment by xuanxu** [comment]
@whedon set 10.5281/zenodo.3831586 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3831586" target="_blank">10.5281/zenodo.3831586</a> is the archive.

**Comment by xuanxu** [comment]
@whedon set v1.0.0 as version

**Comment by whedon** [comment]
OK. v1.0.0 is the version.

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02101/joss.02101/10.21105.joss.02101.pdf)

**Comment by xuanxu** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1448

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1448, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1262338226251075584) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1449
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02101](https://doi.org/10.21105/joss.02101)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@christinahedges, @aureliocarnero - many thanks for your reviews here and to @xuanxu for editing this submission ✨ 

@noraeisner :wave: hope you are well! Your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02101/status.svg)](https://doi.org/10.21105/joss.02101)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02101">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02101/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02101/status.svg
   :target: https://doi.org/10.21105/joss.02101
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02101/status.svg)](https://doi.org/10.21105/joss.02101)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
